src/downshift.js

```diff
@@ -52,15 +52,15 @@ class Downshift extends Component {
     itemCount: PropTypes.number,
     id: PropTypes.string,
     environment: PropTypes.shape({
-      addEventListener: PropTypes.func.isRequired,
-      removeEventListener: PropTypes.func.isRequired,
+      addEventListener: PropTypes.func,
+      removeEventListener: PropTypes.func,
       document: PropTypes.shape({
         createElement: PropTypes.func.isRequired,
         getElementById: PropTypes.func.isRequired,
         activeElement: PropTypes.any.isRequired,
         body: PropTypes.any.isRequired,
-      }).isRequired,
-      Node: PropTypes.func.isRequired,
+      }),
+      Node: PropTypes.func,
```

Comment:
Note that these changes as is would mean the prop types and Typescript types will diverge slightly. 

I typed out some potential options but realized that the most accurate is probably just to have the types reflect the reality that `environment` can either be a full `Environment` (with required props) or an empty object. Will make that change.